### PR TITLE
[DROOLS-6395] PrometheusMetricsDroolsListener is repeatedly added to …

### PR DESCRIPTION
**Pull request by tkobayas**
…the same ksession instance

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-6395

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Review by lucamolteni — APPROVED** [review]

**Comment by tkobayas** [comment]
Karaf tests failed with other PR jobs around the time so not related to this PR. (and Build Chain / Maven Build (11) was successful)

**Comment by tkobayas** [comment]
Jenkins retest this

**Comment by tkobayas** [comment]
@sutaakar Please review, thanks!

Karaf tests failed with other PR jobs around the time so not related to this PR. (and Build Chain / Maven Build (11) was successful)

**Review by sutaakar — APPROVED** [review]
Looks good to me, works as expected

File: kie-server-parent/kie-server-services/kie-server-services-drools/src/main/java/org/kie/server/services/drools/DroolsKieSessionLookupHandler.java [hunk 1]
```diff
@@ -65,7 +65,9 @@ public void postLookupKieSession(String kieSessionId, KieContainerInstance conta
             //default handler
             PrometheusMetricsDroolsListener listener = new PrometheusMetricsDroolsListener(PrometheusKieServerExtension.getMetrics(),
                     kieSessionId, containerInstance);
-            eventManager.addEventListener(listener);
+            if (!eventManager.getAgendaEventListeners().contains(listener)) {
```
Comment (mariofusco) [inline kie-server-parent/kie-server-services/kie-server-services-drools/src/main/java/org/kie/server/services/drools/DroolsKieSessionLookupHandler.java:68]: Sorry, I know I'm being fussy here, but I'm not sure if this is the best way to fix the problem. In particular I don't think that a listener should implement equals/hashCode and moreover doing so in most cases you're just creating a listener instance with the only purpose to compare it and then throw it away.

I assume that it makes sense to only have one `PrometheusMetricsDroolsListener` for each and every session and if that's true I'd prefer to replace this fix with something like the following
```

if (eventManager.getAgendaEventListeners().stream().noneMatch(PrometheusMetricsDroolsListener.class::isInstance)) {
    PrometheusMetricsDroolsListener listener = new PrometheusMetricsDroolsListener(PrometheusKieServerExtension.getMetrics(),
            kieSessionId, containerInstance);
    eventManager.addEventListener(listener);
}
```

**Review by mariofusco — APPROVED** [review]

File: kie-server-parent/kie-server-services/kie-server-services-drools/src/main/java/org/kie/server/services/drools/DroolsKieSessionLookupHandler.java (hunk 1 above)
Comment (tkobayas) [reply kie-server-parent/kie-server-services/kie-server-services-drools/src/main/java/org/kie/server/services/drools/DroolsKieSessionLookupHandler.java]: @mariofusco Thank you! You are absolutely right. Fixed.

**Comment by tkobayas** [comment]
Jenkins retest this

**Pull request merged**